core/include/detray/builders/cuboid_portal_generator.hpp

```diff
@@ -86,15 +87,15 @@ class cuboid_portal_generator final
                     typename detector_t::geometry_context ctx = {})
         -> dindex_range override {
 
-        using point3_t = typename detector_t::point3_type;
-        using vector3_t = typename detector_t::vector3_type;
+        using point3_t = dpoint3D<algebra_type>;
+        using vector3_t = dvector3D<algebra_type>;
```

Comment:
Is there actually a point to making these changes? 🤔 I.e. do we foresee to be able to use let's say a detector holding `float` constants, with calculations using `double` values?

If yes, then let's do all this. If no, if that will never happen, then we might as well continue taking the types from `detector_t`. 🤔

Reply:
I don't like having everything depend directly on the detector, when the detector does not really need these types itself.